docs: content for module 5 (#8)
asyncapi/learning-paths: 100-level-AsyncAPI-Intro/step5.md

> In this learning journey, you explore Async API, a powerful tool for crafting asynchronous APIs. The first module discusses what AsyncAPI is and why to use Async API and explores real-world use cases where AsyncAPI is a valuable asset.

Reviewer: Please fix how you wrote AsyncAPI.

> Building on the basics, module two transitions into understanding Event-driven Architecture(EDA), showcasing how AsyncAPI fuels the design of both EDA and microservices.

Reviewer: ```suggestion
Building on the basics, module two transitions into understanding Event-driven Architecture (EDA), showcasing how AsyncAPI fuels the design of both EDA and microservices.
```

> The third module explores the AsyncAPI specification. You learn about the structure of AsyncAPI and its different components, such as info, channels, components, and tags. You also learn to define operations, messages, and schemas, and the request/reply pattern.

Reviewer 2: WDYT? 
```suggestion
The third module explores the AsyncAPI specification. You learned about the structure of an AsyncAPI document and its different components, such as info, channels, operations, and messages. You also learned to implement the request/reply pattern.
```

Author: IMO: we try to use active voice while documenting, but i understand the changes requested.

> The fourth module focuses on hands-on experience with exercises for writing AsyncAPI code and documents from scratch. It also introduces AsyncAPI Studio for validating your documents.

Reviewer 2: ```suggestion
The fourth module focuses on hands-on experience with exercises for writing AsyncAPI documents from scratch and generating code from them. It also introduces AsyncAPI Studio for validating your documents.
```

Reviewer 2: BTW, nothing about generating code is explained atm in this module. See https://github.com/asyncapi/learning-paths/pull/4#issuecomment-1952469968

Author: I would have to ask the author of module 4 for this, let me ping her.

Reviewer: Yes, Sergio is correct that Module 4 lacks the "code generation" content. (which will be addressed in that PR)

However, this PR is for module 5 and Hrideysh is correct in noting that module 4 should cover code generation. IOW, nothing for Hridyesh to correct, it's Mahfuza's work that needs correction. 👍🏻

> In this learning journey, you explore AsyncAPI, a powerful tool for crafting asynchronous APIs. The first module discusses what AsyncAPI is and why to use AsyncAPI and explores real-world use cases where AsyncAPI is a valuable asset.
> 
> Building on the basics, module two transitions into understanding Event-driven Architecture (EDA), showcasing how AsyncAPI fuels the design of both EDA and microservices.
> 
> The third module explores the AsyncAPI specifications. You learn about the structure of AsyncAPI and its different components, such as info, channels, components, and tags. You also learn to define operations, messages, and schemas and grasp the request/reply patterns within AsyncAPI.
> 
> The fourth module focuses on hands-on experience by learning to write AsyncAPI code and documents from scratch. The module also introduces tools to validate these documents and practical exercises to solidify your learning.

Reviewer: ```suggestion
In this intergalactic journey, you explored AsyncAPI, a powerful specification for crafting asynchronous APIs. The first module discusses what AsyncAPI is and why to use AsyncAPI, exploring real-world AsyncAPI use cases.

Building on the basics, module two transitions into understanding Event-driven Architecture (EDA), showcasing how AsyncAPI fuels the design of both EDA and microservices.

The third module explores the AsyncAPI specification. You learn about the structure of AsyncAPI and its different components, such as info, channels, components, and tags. You also learn to define operations, messages, and schemas, and the request/reply pattern.

The fourth module focuses on hands-on experience with exercises for writing AsyncAPI code and documents from scratch. It also introduces AsyncAPI Studio for validating your documents.
```

## AsyncAPI tools
Ready to explore more of the AsyncAPI world? Check out the AsyncAPI tool ecosystem!

- AsyncAPI Studio helps you design and create a blueprint for your asynchronous API. 
- AsyncAPI CLI (Command-Line Interface) helps you run AsyncAPI using text commands in a terminal.
- AsyncAPI Generator helps you plan and build the initial structure of your project. 

> - And more!

Reviewer 2: Should we add links?

Author: We could, let me add them

Author: Should i link it to different GitHub repos or documentation?

Reviewer: Yes, @kakabisht, we should add links to the tools. Let's link to the tool repository because that is a better user experience. (unless someone else thinks differently? open to suggestions)

> Async CLI (Command-Line Interface) helps you run AsyncAPI using text commands in a terminal. For example, you type “Bake the cake” in the AsyncCLI, and the CLI bakes a cake for you.

Reviewer: Please fix how you wrote AsyncAPI CLI.

> 
> AsyncAPI Studio helps you design and create a blueprint for your asynchronous API. For example, drawing a map displaying how to exchange messages between two teams.
> 
> AsyncAPI CLI (Command-Line Interface) helps you run AsyncAPI using text commands in a terminal. For example, you type “Bake the cake” in the AsyncAPI CLI, and the AsyncAPI CLI bakes a cake for you.
> 
> Generator helps you plan and build the initial structure of your project. For example, It's like having an assistant who sets up the starting point of your birthday party.

Reviewer: ```suggestion
Ready to explore more of the AsyncAPI world? Check out the AsyncAPI tool ecosystem!

- AsyncAPI Studio helps you design and create a blueprint for your asynchronous API. 
- AsyncAPI CLI (Command-Line Interface) helps you run AsyncAPI using text commands in a terminal.
- AsyncAPI Generator helps you plan and build the initial structure of your project. 
- And more!
```

## Extra resources

* [AsyncAPI Documentation](https://www.asyncapi.com/docs/): The official documentation provides comprehensive examples and context for learning AsyncAPI.

> * [AsyncAPI GitHub Repository](https://github.com/asyncapi#-join-asyncapi-community): Join the AsyncAPI community and start contributing.

Reviewer 2: ```suggestion
* [AsyncAPI Community GitHub Repository](https://github.com/asyncapi#-join-asyncapi-community): Join the AsyncAPI community and start contributing.
```

> * [AsyncAPI Documentation](https://www.asyncapi.com/docs/): The official documentation provides comprehensive information, examples, and guides for understanding and using AsyncAPI.

Reviewer: ```suggestion
* [AsyncAPI Documentation](https://www.asyncapi.com/docs/): The official documentation provides comprehensive examples and context for learning AsyncAPI.
```

> * [AsyncAPI GitHub Repository](https://github.com/asyncapi#-join-asyncapi-community): AsyncAPI GitHub README - Join the AsyncAPI community and start contributing.

Reviewer: ```suggestion
* [AsyncAPI GitHub Repository](https://github.com/asyncapi#-join-asyncapi-community): Join the AsyncAPI community and start contributing.
```

1. Introduction to AsyncAPI:
   1. Explore the fundamentals of AsyncAPI.
   1. Understand the reasons for using AsyncAPI.
   1. Discover real-world use cases demonstrating the practical applications of AsyncAPI.
1. Event-Driven Architectures:
   1. Dive into the concept of event-driven architectures.
   1. Learn how AsyncAPI can be a powerful tool in designing Event-driven systems
   1. Understand the role of AsyncAPI in crafting APIs for microservices.
1. AsyncAPI Specification:
   1. Grasp the structure of an AsyncAPI document.
   1. Explore various components such as info, channels, components, and tags.
   1. Learn to define operations, messages, and schemas, including the request/reply pattern.
1. Creating and Validating AsyncAPI Code and Documents:
   1. Gain practical skills in writing AsyncAPI code and documents from scratch.
   1. Utilize AsyncAPI tools to validate the created documents.
   1. Engage in hands-on exercises to reinforce your understanding and skills.

> ## AsyncAPI toolkit

Reviewer: ```suggestion
## AsyncAPI tools
```

Reviewer: Introduce the topic of AsyncAPI having tools.

Reviewer: let's remove metaphors for the tools section

> There are different creative tools in your AsyncAPI toolkit:

Reviewer: ```suggestion
```

AsyncAPI Studio helps you design and create a blueprint for your asynchronous API. For example, you are planning a birthday party and drawing a map displaying how messages move from one friend to another.
Async CLI (Command-Line Interface) helps you run AsyncAPI using text commands in a terminal. For example, you type “Bake the cake” in the AsyncCLI, and the CLI bakes a cake for you.
Generator helps you plan and build the initial structure of your project. For example, It's like having an assistant who sets up the starting point of your birthday party.
AsyncAPI Studio helps you draw your API plans, the CLI checks if everything's in order, and the Generator magically creates the starting point for your projects.

## Extra resources

- [AsyncAPI Documentation](https://www.asyncapi.com/docs/): The official documentation provides comprehensive information, examples, and guides for understanding and using AsyncAPI.

> - [AsyncAPI GitHub Repository](https://github.com/asyncapi/asyncapi): AsyncAPI GitHub - Explore the AsyncAPI GitHub repository for code samples, issues, and community contributions.

Reviewer: ```suggestion
- [AsyncAPI GitHub Repository](https://github.com/asyncapi#-join-asyncapi-community): AsyncAPI GitHub README - Join the AsyncAPI community and start contributing.
```

> - [AsyncAPI YouTube Tutorials](https://www.youtube.com/c/AsyncAPI): AsyncAPI YouTube - Access video tutorials covering various aspects of AsyncAPI, enhancing your learning experience.

Reviewer: ```suggestion
- [AsyncAPI YouTube](https://www.youtube.com/c/AsyncAPI): Watch AsyncAPI community meetings and learning content on our YouTube channel.
```
